pkg/kn/commands/plugin/plugin.go

```diff
@@ -52,3 +52,20 @@ func BindPluginsFlagToViper(cmd *cobra.Command) {
 	viper.SetDefault("plugins-dir", commands.Cfg.DefaultPluginDir)
 	viper.SetDefault("lookup-plugins", false)
 }
+
+// AllowedExtensibleCommandGroups the list of command groups that can be
+// extended with plugins, e.g., a plugin named `kn-source-kafka` for Kafka
+// event sources is allowed. This is defined as a fixed [...]string since
+// cannot defined Golang []string constants
+var AllowedExtensibleCommandGroups = [...]string{"source"}
```

Comment:
I think the command group to extend should be configurable outside the kn, i.e config.
OR
It shouldn't mind extending command groups if the plugin command dont clash.

Reply:
Not sure about that since we would not want extensions of core commands

Reply:
as per today’s discussion, please merge this PR and I will open an issue with your suggestion with more details and open a 24 hours vote. Thanks @navidshaikh and @rhuss